packages/beacon-node/src/chain/archiveStore/archiveStore.ts

```diff
@@ -30,8 +32,17 @@ export class ArchiverStore {
 
     if (!opts.disableArchiveOnCheckpoint) return;
 
+    const mediator = new MediatorQueueObserver({
+      // As archive logic is splitted into separate observers, and each observer event is treated
+      // as it's own queue item. To keep matching with existing behavior multiplying old
+      // queue length with number of observers
+      maxQueueLength: PROCESS_FINALIZED_CHECKPOINT_QUEUE_LEN * 5,
+      signal,
+      logger,
+    });
```

Comment:
Previous implementation was making sure that each event handler are called in sequence. But because each had it's own queue we were not sure if one handler is called after other one is finished. 

This implementation makes sure that there is only one queue used for all observers. So each must be invoked in sequence as well as wait for other to be finished first.

Reply:
This feels like a big API issue.  If this is the case it feels like the original way where one class observed the events and then a handler on that class called the different operations in the correct sequence and with the correct timing (async for some parts or requiring sync between steps for others)